File: implementations/flutter/design_patterns_flutter/lib/patterns/structural/bridge/inventory_report.dart
import 'abstraction.dart';
import 'domain/report_data.dart';

class Product {
  final String sku;
  final String name;
  final int stock;
  final int min;
  const Product({
    required this.sku,
    required this.name,
    required this.stock,
    required this.min,
  });
}

class InventoryReport extends Report {
  final List<Product> items;
  InventoryReport(super.exporter, this.items);

  @override
  Future<ReportData> build() async {
    final rows = items
        .map(
          (p) => {
            'sku': p.sku,
            'name': p.name,
            'stock': p.stock,
            'min': p.min,
            'value': p.stock * p.min,
          },
        )
        .toList();
    return ReportData(title: 'Inventory Report', rows: rows);
  }
}

File: implementations/react-native/design-patterns-rn/src/patterns/structural/bridge/inventory-report.ts
import { Report } from './abstraction';
import type { Product } from './interfaces/product';
import type { Exporter } from './implementor';
import type { ReportData } from './interfaces/report-data';

export class InventoryReport extends Report {
  private items: Product[];
  constructor(exporter: Exporter, items: Product[]) { super(exporter); this.items = items; }

  async build(): Promise<ReportData> {
    const rows = this.items.map((p) => ({
      sku: p.sku,
      name: p.name,
      stock: p.stock,
      min: p.min,
      value: p.stock * p.min,
    }));
    return { title: 'Inventory Report', rows };
  }
}
